x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/AbstractProfilingPersistenceManager.java

```diff
@@ -282,80 +200,4 @@ public void onFailure(Exception ex) {
             }, consumer);
         });
     }
-
-    private int getVersionField(Index index, Map<String, Object> meta, String fieldName) {
-        Object value = meta.get(fieldName);
-        if (value instanceof Integer) {
-            return (int) value;
-        }
-        if (value == null) {
-            logger.warn("Metadata version field [{}] of index [{}] is empty.", fieldName, index);
-            return -1;
-        }
-        logger.warn("Metadata version field [{}] of index [{}] is [{}] (expected an integer).", fieldName, index, value);
-        return -1;
-    }
-
-    protected static final class IndexState<T extends ProfilingIndexAbstraction> {
-        private final T index;
-        private final Index writeIndex;
-        private final Status status;
-        private final List<Migration> pendingMigrations;
-
-        IndexState(T index, Index writeIndex, Status status) {
-            this(index, writeIndex, status, null);
-        }
-
-        IndexState(T index, Index writeIndex, Status status, List<Migration> pendingMigrations) {
-            this.index = index;
-            this.writeIndex = writeIndex;
-            this.status = status;
-            this.pendingMigrations = pendingMigrations;
-        }
-
-        public T getIndex() {
-            return index;
-        }
-
-        public Index getWriteIndex() {
-            return writeIndex;
-        }
-
-        public Status getStatus() {
-            return status;
-        }
-
-        public List<Migration> getPendingMigrations() {
-            return pendingMigrations;
-        }
-    }
-
-    enum Status {
-        CLOSED(false),
-        UNHEALTHY(false),
-        NEEDS_CREATION(true),
-        NEEDS_VERSION_BUMP(true),
-        UP_TO_DATE(false),
-        NEEDS_MAPPINGS_UPDATE(true);
-
-        /**
-         * Whether a status is for informational purposes only or whether it should be acted upon and may change cluster state.
-         */
-        private final boolean actionable;
-
-        Status(boolean actionable) {
-            this.actionable = actionable;
-        }
-    }
-
-    /**
-     * An index that is used by Universal Profiling.
-     */
-    interface ProfilingIndexAbstraction {
```

Comment:
This interface has been moved to the top-level interface `ProfilingIndexAbstraction`.